price_list.py

```diff
@@ -0,0 +1,75 @@
+# This file is part of Tryton.  The COPYRIGHT file at the top level of
+# this repository contains the full copyright notices and license terms.
+from trytond.pool import PoolMeta, Pool
+
+
+class PriceList(metaclass=PoolMeta):
+    __name__ = 'product.price_list'
+
+    def get_context_formula(self, party, product, unit_price, quantity, uom,
+            pattern=None):
+        pool = Pool()
+        Product = pool.get('product.product')
+        res = super().get_context_formula(party, product, unit_price, quantity,
+            uom, pattern)
+
+        if not product:
```

Comment:
Entenc que si no hi ha producte retornem el res del super i ja està no?
perquè busquem el primer ?

Reply:
Quan escrivim una fórmula nova o modifiquem una fórmula de les tarifes, tryton la comprova.
Aquest "if not product" només salta en aquesta comprovació. Si no, no és vàlida la formula.

Podem veure altres llocs on s'utilitza el mateix com per exemple:
https://github.com/NaN-tic/trytond-product_price_list_formula/blob/main/price_list.py#L49